walg_archive.c

```diff
@@ -74,8 +84,21 @@ static bool
 walg_archive_configured(void)
 {
 	fd = set_connection();
-	char *check_message = "CHECK";
-	if (send(fd, check_message, strlen(check_message)*sizeof(char), 0 ) == -1)
+
+	char message_type[] = "C";
+	char message_body[] = "CHECK";
+	uint8 message_len = sizeof(message_body) + 1;
+
+	char *p = palloc(sizeof(char)*message_len);
```

Comment:
do not forget to use `pfree` to free the allocated memory, otherwise it is a memory leak :)